app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :user, inverse_of: :posts, foreign_key: :user_id, counter_cache: true
+  has_many :comments, dependent: :destroy, inverse_of: :post, foreign_key: :post_id
+  has_many :likes, dependent: :destroy, inverse_of: :post, foreign_key: :post_id
+
+  def update_post_counter
+    user.update(post_counter: user.posts.count)
+  end
```

Comment:
- [ ] You should increment the post counter for the user in this method and call the method after saving so when a user creates a post the counter should be incremented automatically. Please fix this issue.
  - You can consider the below code.
```ruby
after_save :update_post_counter
  def update_post_counter
    user.increment!(:post_counter)
  end